pg-extend/src/pg_fdw.rs

```diff
@@ -242,8 +242,9 @@ impl<T: ForeignData> ForeignWrapper<T> {
         t
     }
 
-    unsafe fn tupdesc_attrs(tupledesc: &pg_sys::TupleDesc) -> &[pg_sys::Form_pg_attribute] {
-        let tupledesc = *tupledesc;
+    // This isn't quite 'static, but it will last for the duration of the
+    // callback, which is good enough
+    unsafe fn tupdesc_attrs(tupledesc: pg_sys::TupleDesc) -> &'static [pg_sys::Form_pg_attribute] {
```

Comment:
similar comment about the param, the static lifetime wouldn't be needed in that case. And won't this have issues related to when the data is freed? I'm guessing it's relying on the fact that the pg allocator is connection based on free...

Reply:
The reason that I consider this to be static is that it's allocated outside the borrow checker's perview (inside the postgres internals) and deallocated outside as well (as the result of a callback). We always make copies when we use anything inside Rust code, so from Rust's perspective it's static in that it will never not be present.

Reply:
I see. I'm not sure I completely agree, but I see what you mean. I just worry about abuse, i.e. sticking it in a thread local or something crazy ;)

but we can merge as is and see if there are improvements as necessary.

Reply:
Hmm. one thing I could do is take a reference to `tupleDesc` as opposed to `TupleDesc`, which would fix the redundancy.

Reply:
@bluejekyll updated!